app/javascript/App/bundles/CoinShow/Fundamentals.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react'
+import Icon from '~/bundles/common/components/Icon'
+import { Grid, withStyles, createStyles } from '@material-ui/core'
+import {
+  formatValueWithCurrency,
+  formatValue,
+} from '~/bundles/common/utils/numberFormatters'
+
+interface Props {
+  coinObj: any
+  currency: string
+  classes: any
+}
+
+const styles = (theme) =>
+  createStyles({
+    title: {
+      color: 'rgba(0,0,0,0.38)',
+      marginBottom: '4px',
+      fontSize: '14px',
+      lineHeight: '22px',
+    },
+    value: {
+      fontWeight: 'bold',
+      color: '#555',
+      marginBottom: '4px',
+      fontSize: '14px',
+      lineHeight: '22px',
+    },
+  })
+
+class Fundamentals extends React.Component<Props, {}> {
+  public render() {
+    const { coinObj, classes } = this.props
+    const currency = this.props.currency.toUpperCase()
+    const currencyKey = currency.toLowerCase()
+    const {
+      market_cap,
+      change24h,
+      change7d,
+      available_supply,
+      total_supply,
+      max_supply,
+      symbol,
+    } = coinObj
+    const hasMarketCap =
+      typeof market_cap === 'object' && market_cap[currencyKey]
```

Comment:
For future reference. stylistically, should we just use lodash wherever possible or avoid it unless necessary?
I'm wondering because, for front-end, it might also be preferable to only import the necessary lodash methods rather than always having the full library.

Reply:
I think we should use lodash where possible just so we manage less code and ultimately avoid having to write annoying things like `typeof ... === 'undefined'`. I have tried to optionally pull in specific lodash methods but I because tend to rely on lodash a lot so it ended up being quite tedious so I like just having the full library there.

What do you guys think?

Reply:
I'm happy with using lodash methods where possible to simplify code. Seems with [babel/webpack](https://www.blazemeter.com/blog/the-correct-way-to-import-lodash-libraries-a-benchmark) you can achieve the same bundle size as individually importing lodash methods anyways.

Reply:
I'm totally fine with just using lodash.

There might be some automated performance optimizations we can get via https://github.com/GoogleChromeLabs/webpack-libs-optimizations#lodash